src/snowflake/connector/query_context_cache.py

```diff
@@ -180,91 +169,115 @@ def _get_elements(self) -> set[QueryContextElement]:
-
-                stream.seek(0)
-                # use same encoding use on jdbc driver
-                data = b64encode(stream.read()).decode("iso-8859-1")
-                stream.close()
+                data = {}
+                if(size >= 1):
+                    data = {
+                        "entries": [
+                            {
+                                "id": id_vals[i],
+                                "timestamp": timestamp_vals[i],
+                                "priority": priority_vals[i],
+                                "context": context_vals[i], 
+                            }
+                            for i in range(0, size)
+                        ]
+                    }
+                # Serialize the data to JSON
+                serialized_data = json.dumps(data)
 
                 logger.debug(
-                    f"serialize_to_arrow_base64(): data to send to server {data}"
+                    f"serialize_to_json(): data to send to server {serialized_data}"
                 )
-                return data
+
+                return serialized_data
             except Exception as e:
-                logger.debug(f"serialize_to_arrow_base64(): Exception {e}")
-                return None
+                logger.debug(f"serialize_to_json(): Exception {e}")
+                return None   
+
+    def deserialize_json_string(self, json_string: str) -> None:
+        with self._lock:
+            logger.debug(
+                f"deserialize_from_json() called: data from server: {json_string}"
+            )
+            self.log_cache_entries()
+            
+            if json_string is None or len(json_string) == 0:
+                self.clear_cache()
+                logger.debug("deserialize_from_json() returns")
+                self.log_cache_entries()
+                return
+            
+            try:
+                data = json.loads(json_string)
+                # Deserialize the entries. The first entry with priority is the main entry. On JDBC side, 
+                # we save all entries into one list to simplify the logic. An example JSON is:
+                # {
+                #   "entries": [
+                #    {
+                #     "id": 0,    
+                #     "read_timestamp": 123456789,
+                #     "priority": 0,
+                #     "context": "base64 encoded context"
+                #    },
+                #     {
+                #       "id": 1,
+                #       "read_timestamp": 123456789,
+                #       "priority": 1,
+                #       "context": "base64 encoded context"
+                #     },
+                #     {
+                #       "id": 2,
+                #       "read_timestamp": 123456789,
+                #       "priority": 2,
+                #       "context": "base64 encoded context"
+                #     }
+                #   ]
+                if("entries" in data):
```

Comment:
```suggestion
                #   ]
                # }
                if("entries" in data):
```
nit: closing the left curly braces